pkg/resources/otoroshi/crud.go

```diff
@@ -0,0 +1,216 @@
+package otoroshi
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"go.clever-cloud.com/terraform-provider/pkg"
+	"go.clever-cloud.com/terraform-provider/pkg/helper"
+	"go.clever-cloud.com/terraform-provider/pkg/provider"
+	"go.clever-cloud.com/terraform-provider/pkg/tmp"
+)
+
+func (r *ResourceOtoroshi) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	tflog.Debug(ctx, "ResourceOtoroshi.Configure()")
+
+	if req.ProviderData == nil {
+		return
+	}
+
+	provider, ok := req.ProviderData.(provider.Provider)
+	if ok {
+		r.cc = provider.Client()
+		r.org = provider.Organization()
+	}
+}
+
+func (r *ResourceOtoroshi) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	otoroshi := helper.PlanFrom[Otoroshi](ctx, req.Plan, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	addonsProvidersRes := tmp.GetAddonsProviders(ctx, r.cc)
+	if addonsProvidersRes.HasError() {
+		resp.Diagnostics.AddError("failed to get add-on providers", addonsProvidersRes.Error().Error())
+		return
+	}
+
+	addonsProviders := addonsProvidersRes.Payload()
+
+	provider := pkg.LookupAddonProvider(*addonsProviders, "otoroshi")
+	if provider == nil {
+		resp.Diagnostics.AddError("Otoroshi provider doesn't exist", fmt.Sprintf("available providers are: %s", strings.Join(pkg.AddonProvidersAsList(*addonsProviders), ", ")))
+		return
+	}
+	plan := provider.Plans[0]
+
+	addonReq := tmp.AddonRequest{
+		Name:       otoroshi.Name.ValueString(),
+		Plan:       plan.ID,
+		ProviderID: provider.ID,
+		Region:     otoroshi.Region.ValueString(),
+	}
+
+	if !otoroshi.Version.IsNull() && !otoroshi.Version.IsUnknown() {
+		addonReq.Options = map[string]string{
+			"version": otoroshi.Version.ValueString(),
+		}
+	}
+
+	res := tmp.CreateAddon(ctx, r.cc, r.org, addonReq)
+	if res.HasError() {
+		resp.Diagnostics.AddError("failed to create Otoroshi add-on", res.Error().Error())
+		return
+	}
+	addonRes := res.Payload()
+
```

Comment:
At this step, you should store at least the addon real ID, and save the state, this prevent resource leak

Reply:
(L78 & L79)